mapadroid/worker/WorkerMITM.py

```diff
@@ -112,17 +115,28 @@ def _move_to_location(self):
         else:
             self.logger.info("main: Walking...")
             self._transporttype = 1
+            time_before_walk = math.floor(time.time())
             self._communicator.walk_from_to(self.last_location, self.current_location, speed)
-            # the time we will take as a starting point to wait for data...
-            cur_time = math.floor(time.time())
+            # We need to roughly estimate when data could have been available, just picking half way for now, distance
+            # check should do the rest...
+            timestamp_to_use = math.floor(time.time())
+            if timestamp_to_use - 10 < time_before_walk:
```

Comment:
We should use a variable instead of a magic number in case we need to change this later.